src/Microsoft.AspNetCore.Razor.Language/DefaultRazorProjectFileSystem.cs

```diff
@@ -49,6 +49,10 @@ public override RazorProjectItem GetItem(string path)
             var absolutePath = NormalizeAndEnsureValidPath(path);
 
             var file = new FileInfo(absolutePath);
+            if (!absolutePath.StartsWith(absoluteBasePath))
+            {
+                throw new InvalidOperationException($"The file '{file.FullName}' is not a descendent of the base path '{absoluteBasePath}'.");
+            }
```

Comment:
These cases would explode at line 57 currently. Hit this while testing something new. All tests passed when I added this validation.